trie/nodeset.go

```diff
@@ -124,36 +112,61 @@ func NewNodeSetWithDeletion(owner common.Hash, paths [][]byte, prev [][]byte) *N
 }
 */
 
+// forEachWithOrder iterates the dirty nodes with the specified order.
+// If topToBottom is true:
+//
+//	then the order of iteration is top to bottom, left to right.
+//
+// If topToBottom is false:
+//
+//	then the order of iteration is bottom to top, right to left.
+func (set *NodeSet) forEachWithOrder(topToBottom bool, callback func(path string, n *nodeWithPrev)) {
```

Comment:
Yep. 
If `topToBottom` is true, nodes are iterated from root to leaves, from left to right;
If `topToBottom` is false, nodes are iterated from leaves to root, from right to left;

For us, `topToBottom` is used, with which we always flush children first then parent. Regarding the order between siblings, it's not strict that we must go from left to right, or right to left. It will not break the guarantee that: **if parent node is present, the corresponding subtrie is present as well**.

It's an idea from @holiman, I will just cc him here for more eyes.